src/lib/route_rule.rs

```diff
@@ -0,0 +1,206 @@
+            FR_ACT_TO_TBL => Self::Table,
+            FR_ACT_GOTO => Self::Goto,
+            FR_ACT_NOP => Self::Nop,
+            FR_ACT_RES3 => Self::Res3,
+            FR_ACT_RES4 => Self::Res4,
+            FR_ACT_BLACKHOLE => Self::Blackhole,
+            FR_ACT_UNREACHABLE => Self::Unreachable,
+            FR_ACT_PROHIBIT => Self::Prohibit,
+            _ => Self::Other(d),
+        }
+    }
+}
+
+impl Default for RuleAction {
+    fn default() -> Self {
+        Self::Unknown
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Default)]
+pub struct RouteRule {
+    pub action: RuleAction,
+    pub address_family: AddressFamily,
+    pub flags: u32,
+    pub table: u32,
+    pub tos: u8,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dst: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub src: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub iif: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub oif: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub goto: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub priority: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fw_mark: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fw_mask: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub mask: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub flow: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub tun_id: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub suppress_ifgroup: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub suppress_prefix_len: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub protocol: Option<RouteProtocol>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ip_proto: Option<AddressFamily>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub src_port_range: Option<Vec<u8>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dst_port_range: Option<Vec<u8>>,
+}
+
+pub(crate) fn get_route_rules() -> Result<Vec<RouteRule>, NisporError> {
+    Ok(Runtime::new()?.block_on(_get_route_rules())?)
+}
+
+async fn _get_route_rules() -> Result<Vec<RouteRule>, NisporError> {
+    let mut rules = Vec::new();
+    let (connection, handle, _) = new_connection()?;
+    tokio::spawn(connection);
+
+    let mut links = handle.rule().get(IpVersion::V6).execute();
+    while let Some(rt_msg) = links.try_next().await? {
+        rules.push(get_rule(rt_msg)?);
+    }
+    let mut links = handle.rule().get(IpVersion::V4).execute();
+    while let Some(rt_msg) = links.try_next().await? {
+        rules.push(get_rule(rt_msg)?);
+    }
+    Ok(rules)
+}
+
+fn get_rule(rule_msg: RuleMessage) -> Result<RouteRule, NisporError> {
```

Comment:
This function never raise `NisporError`, no need to use `Result<>`.

Reply:
Done, thank you!